solana/modules/wormhole-anchor-sdk/src/token_bridge/message.rs

```diff
@@ -21,10 +22,60 @@ pub struct TransferWithMeta {
     pub from_address: [u8; 32],
 }
 
-impl AnchorDeserialize for TransferWithMeta {
-    fn deserialize(buf: &mut &[u8]) -> io::Result<Self> {
+impl TransferHeader {
+    #[cfg(test)]
+    pub fn random(rng: &mut impl rand::Rng) -> Self {
+        TransferHeader {
+            amount: rng.gen(),
+            token_address: rng.gen(),
+            token_chain: rng.gen(),
+            to_address: rng.gen(),
+            to_chain: rng.gen(),
+            from_address: rng.gen(),
+        }
+    }
+}
+
+impl Writeable for TransferHeader {
+    fn write<W>(&self, writer: &mut W) -> io::Result<()>
+    where
+        W: io::Write,
+    {
+        let TransferHeader {
+            amount,
+            token_address,
+            token_chain,
+            to_address,
+            to_chain,
+            from_address,
+        } = self;
+
+        PAYLOAD_ID_TRANSFER_WITH_PAYLOAD.write(writer)?;
+        U256::new(*amount).write(writer)?;
+        token_address.write(writer)?;
+        token_chain.write(writer)?;
+        to_address.write(writer)?;
+        to_chain.write(writer)?;
+        from_address.write(writer)?;
+
+        Ok(())
```

Comment:
```suggestion
        from_address.write(writer)
```

That's my preference anyway.

Reply:
I think that when there is a long list that might be expanded, it makes more sense to use the same line for every field for consistency. Also, if we add a field, the git commit makes more sense.